File: bacalhao-a-bras-java/src/main/java/org/jesperancinha/portuguese/recipes/bras/JBakerHelper.java
package org.jesperancinha.portuguese.recipes.bras;

import com.ing.baker.recipe.common.Event;
import com.ing.baker.recipe.common.EventOutputTransformer;
import com.ing.baker.recipe.common.Ingredient;
import com.ing.baker.recipe.scaladsl.Interaction;
import com.ing.baker.types.Value;
import scala.Option;
import scala.collection.JavaConverters;
import scala.collection.Seq;
import scala.collection.immutable.HashMap;
import scala.collection.immutable.HashSet;
import scala.collection.immutable.Map;
import scala.collection.immutable.Set;

import java.util.Arrays;

public class JBakerHelper {

    public static Ingredient createIngredient(String name) {
        return IngredientBuilder.creatIngredient(name);
    }

    public static <T> Seq<T> seq(T... ingredient) {
        return JavaConverters.asScalaBuffer(Arrays.asList(ingredient)).toSeq();
    }

    public static Set<String> setEmptyString() {
        return new HashSet<>();
    }

    public static Set<Set<String>> setOfSetEmptyString() {
        return new HashSet<>();
    }

    public static Map<String, Value> mapOfEmptyValue() {
        return new HashMap<>();
    }

    public static Map<String, String> mapOfEmptyString() {
        return new HashMap<>();
    }

    public static Map<Event, EventOutputTransformer> mapOfEmptyCommonEvent() {
        return new HashMap<>();
    }

    public static <T> Set<T> set(T... ingredient) {
        return JavaConverters.asScalaBuffer(Arrays.asList(ingredient)).toSet();
    }

    public static Interaction createInteraction(String name, Seq<Ingredient> ingredients, Seq<Event> events) {
        return Interaction.apply(
                name, ingredients,
                events, setEmptyString(),
                setOfSetEmptyString(),
                mapOfEmptyValue(),
                mapOfEmptyString(),
                Option.empty(),
                Option.empty(),
                Option.empty(),
                mapOfEmptyCommonEvent(),
                Option.empty());
    }

    public static Interaction createInteraction(String name, Ingredient ingredient, Event event) {
        return Interaction.apply(
                name,
                seq(ingredient),
                seq(event),
                setEmptyString(),
                setOfSetEmptyString(),
                mapOfEmptyValue(),
                mapOfEmptyString(),
                Option.empty(),
                Option.empty(),
                Option.empty(),
                mapOfEmptyCommonEvent(),
                Option.empty());
    }
}

File: filhos-de-natal-kotlin/src/main/kotlin/JBakerHelper.kt
import com.ing.baker.recipe.common.Event
import com.ing.baker.recipe.common.EventOutputTransformer
import com.ing.baker.recipe.common.Ingredient
import com.ing.baker.recipe.scaladsl.Interaction
import com.ing.baker.types.Value
import org.jesperancinha.portuguese.recipes.bras.IngredientBuilder
import scala.Option
import scala.collection.JavaConverters
import scala.collection.Seq
import scala.collection.immutable.HashMap
import scala.collection.immutable.HashSet
import scala.collection.immutable.Map
import scala.collection.immutable.Set
import java.util.*

object JBakerHelper {
    fun createIngredient(name: String?): Ingredient {
        return IngredientBuilder.creatIngredient(name)
    }

    fun <T> seq(vararg ingredient: T): Seq<T> {
        return JavaConverters.asScalaBuffer(Arrays.asList(*ingredient)).toSeq()
    }

    private fun setEmptyString(): Set<String> {
        return HashSet()
    }

    private fun setOfSetEmptyString(): Set<Set<String>> {
        return HashSet()
    }

    private fun mapOfEmptyValue(): Map<String, Value> {
        return HashMap()
    }

    private fun mapOfEmptyString(): Map<String, String> {
        return HashMap()
    }

    private fun mapOfEmptyCommonEvent(): Map<Event, EventOutputTransformer> {
        return HashMap()
    }

    fun <T> set(vararg ingredient: T): Set<T> {
        return JavaConverters.asScalaBuffer(Arrays.asList(*ingredient)).toSet()
    }

    fun createInteraction(name: String?, ingredients: Seq<Ingredient>, events: Seq<Event>): Interaction {
        return Interaction.apply(
                name, ingredients,
                events, setEmptyString(),
                setOfSetEmptyString(),
                mapOfEmptyValue(),
                mapOfEmptyString(),
                Option.empty(),
                Option.empty(),
                Option.empty(),
                mapOfEmptyCommonEvent(),
                Option.empty())
    }
}
